File: LogicalThinkingPatterns/NumberCrownPattern/NumberCrown.ts
class NumberCrown {
  printPattern(n: number): void {
    for (let i = 1; i <= n; i++) {
      let pattern = "";
      for (let j = 1; j <= i; j++) {
        pattern += j + " ";
      }
      for (let k = 0; k < 4 * (n - i); k++) {
        pattern += " ";
      }
      for (let l = i; l >= 1; l--) {
        pattern += l + " ";
      }
      console.log(pattern);
    }
  }
}

const crown = new NumberCrown();
crown.printPattern(5);

File: LogicalThinkingPatterns/NumberCrownPattern/NumberCrown.py
class NumberCrown:
    def printPattern(self,n):
        for i in range(1, n+1):
            pattern = ""
            for j in range(1, i+1):
                pattern += str(j)+" "
            for k in range(0, 4*(n-i)):
                pattern += " "
            for l in range(i, 0, -1):
                pattern += str(l)+" "
            print(pattern)
    
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

crown = NumberCrown()
crown.run()